### PR TITLE
♻️🐛 Geo 20180628 tidy

### PR DESCRIPTION
**Pull request by jpettitt**
Fix an undefined in a message and refactor checkCountries_ to use array.reduce

### CODE REVIEW

File: extensions/amp-geo/0.1/amp-geo.js [hunk 1]
```diff
@@ -224,18 +224,18 @@ export class AmpGeo extends AMP.BaseElement {
...
+      // If it's a valid preset then we expand it.
       if (/^preset-/.test(country)) {
         user().assert(
             isArray(ampGeoPresets[country]),
-            `<${TAG}> preset ${country} not found`);
-        delete countryGroup[idx];
-        presetCountries = presetCountries.concat(ampGeoPresets[country]);
-        return;
+            '<amp-geo> preset ${country} not found');
+        return countries = countries.concat(ampGeoPresets[country]);
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: No need to do the assignment, just return.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 2]
```diff
@@ -224,18 +224,18 @@ export class AmpGeo extends AMP.BaseElement {
    * @return {boolean}
    */
   checkGroup_(countryGroup) {
-    let presetCountries = [];
-    countryGroup = countryGroup.map((country, idx) => {
+    countryGroup = countryGroup.reduce((countries, country) => {
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: Nit: create a new variable. Closure will minify.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 3]
```diff
@@ -224,18 +224,18 @@ export class AmpGeo extends AMP.BaseElement {
...
+      // If it's a valid preset then we expand it.
       if (/^preset-/.test(country)) {
         user().assert(
             isArray(ampGeoPresets[country]),
-            `<${TAG}> preset ${country} not found`);
-        delete countryGroup[idx];
-        presetCountries = presetCountries.concat(ampGeoPresets[country]);
-        return;
+            '<amp-geo> preset ${country} not found');
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: Oh, and `${country}` won't expand here, it's the literal string `'${country}'`.

Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: doh! fixed along with your other comments.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 4]
```diff
@@ -219,25 +219,26 @@ export class AmpGeo extends AMP.BaseElement {
...
    */
   checkGroup_(countryGroup) {
-    countryGroup = countryGroup.reduce((countries, country) => {
+    /** @type {!Array<string>} */
+    const expandedGroup = countryGroup.reduce((countries, country) => {
       // If it's a valid preset then we expand it.
       if (/^preset-/.test(country)) {
         user().assert(
             isArray(ampGeoPresets[country]),
-            '<amp-geo> preset ${country} not found');
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: Why are we switching from a template string? It transforms to literally what you wrote.

File: extensions/amp-geo/0.1/amp-geo.js [hunk 5]
```diff
@@ -219,25 +219,26 @@ export class AmpGeo extends AMP.BaseElement {
...
   /**
-   * checkGroup_() does this.country_  match the group
+   * checkGroup_() does this.country_ match the group
+   * after expanding any presets and forceing to lower case.
    * @param {!Array<string>} countryGroup The group to match against
    * @return {boolean}
    */
   checkGroup_(countryGroup) {
-    countryGroup = countryGroup.reduce((countries, country) => {
+    /** @type {!Array<string>} */
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: Nit: Local variable typecasts are unnecessary in 99% of cases.

Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: It complained when I didn't typecast. :-(

Comment (jridgewell) [reply extensions/amp-geo/0.1/amp-geo.js]: Hm, guess it's the 1% of times it's needed.

File: extensions/amp-geo/0.1/amp-geo.js (hunk 4 above)
Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: misread your earlier comment, switched it back.

**Pull request merged**